src/theory/splitter.h

```diff
@@ -0,0 +1,93 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__SPLITTER_H
+#define CVC5__THEORY__SPLITTER_H
+
+#include <math.h>
+
+#include <fstream>
+#include <list>
+#include <sstream>
+
+#include "options/smt_options.h"
+#include "proof/trust_node.h"
+#include "theory/valuation.h"
+
+namespace cvc5 {
+
+class TheoryEngine;
+
+namespace prop {
+class PropEngine;
+}
+
+namespace theory {
+
+class Splitter
+{
+ public:
+  Splitter(TheoryEngine* theoryEngine, prop::PropEngine* propEngine)
+      : d_numPartitions(options::computePartitions()),
+        d_numChecks(0),
+        d_numPartitionsSoFar(0),
+        d_partitionFile(options::writePartitionsToFileName())
+  {
```

Comment:
We tend to have all the function definitions in the source file for anything that is longer than a single line.